### PR TITLE
Activities page

### PR DESCRIPTION
**Pull request by naid**

### CODE REVIEW

File: app/Activity.php [hunk 1]
```diff
@@ -25,4 +25,11 @@ public function getAllUserActivities()
     {
         return Activity::where('user_id', Auth::id())->get();
     }
+
+    public function getUserFolloweeActivities()
+    {
+        return Activity::where('follows.follower_id', Auth::id())
```
Comment (nguyenthanhtung88) [inline app/Activity.php]: Make `Auth::id()` as parameter of this function, e.x: `$userId`
=> It make your function more flexible if in the future, we get information for specific user

File: app/Activity.php [hunk 2]
```diff
@@ -25,4 +25,11 @@ public function getAllUserActivities()
...
+            ->leftJoin('follows', 'follows.followee_id', '=', 'activities.user_id')
```
Comment (nguyenthanhtung88) [inline app/Activity.php]: Why didn't use Eloquent Relationship?

File: app/Activity.php [hunk 3]
```diff
@@ -25,4 +25,11 @@ public function getAllUserActivities()
...
+            ->leftJoin('users', 'users.id', '=', 'follows.followee_id')->get();
```
Comment (nguyenthanhtung88) [inline app/Activity.php]: make `->get()` in new line

File: resources/views/users/view_activities.blade.php [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+                <thead>
+                    <tr>
+                        <th class="col-md-3">Date</th>
+                        <th class="col-md-6">Activity</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    @foreach ($activities as $activity)
+                        <tr>
+                            <td><span>{!! Html::image(
```
Comment (nguyenthanhtung88) [inline resources/views/users/view_activities.blade.php]: Make this code prettier, one suggest for you:
```
<td>
    <span>
        {!! Html::image(
            config()->get('paths.user_image') . $activity->avatar,
            $activity->name,
            ['class' => 'thumbnail'])
        !!}
        {{ $activity->name}}
    </span>
</td>
```

File: database/seeds/ActivitiesTableSeeder.php [hunk 5]
```diff
@@ -38,6 +38,46 @@ public function run()
                 'created_at' => date('Y-m-d G:i:s'),
                 'updated_at' => date('Y-m-d G:i:s'),
             ],
+            [
+                'user_id' => 3,
+                'lesson_id' => 3,
+                'activity' => substr(str_shuffle("0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ , ."), 0, $length),
```
Comment (nguyenthanhtung88) [inline database/seeds/ActivitiesTableSeeder.php]: This function is too long, do you think we need write protected function for doing that, e.x:
```
protected function generateSomething() {
    return substr(str_shuffle("0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ , .");
}
```

File: app/Activity.php (hunk 2 above)
Comment (naid) [inline app/Activity.php]: I read some comment from one of Allan's PR where we can't use some attributes if we use eloquent's 'with' . But if you have a solution, I'm more than open to apply it to this code :-)

File: app/Activity.php [hunk 6]
```diff
@@ -21,8 +21,16 @@ public function lesson()
         return $this->belongsTo(Lesson::class, 'lesson_id');
     }
 
-    public function getAllUserActivities()
+    public function getAllUserActivities($user_id)
```
Comment (nguyenthanhtung88) [inline app/Activity.php]: Make variable name `camelCase`

File: app/Activity.php [hunk 7]
```diff
@@ -21,8 +21,16 @@ public function lesson()
...
     {
-        return Activity::where('user_id', Auth::id())->get();
+        return Activity::where('user_id', $user_id)->get();
+    }
+
+    public function getUserFolloweeActivities($user_id)
```
Comment (nguyenthanhtung88) [inline app/Activity.php]: Use `camelCase` for variable name

File: app/Activity.php [hunk 8]
```diff
@@ -21,8 +21,16 @@ public function lesson()
...
-    public function getAllUserActivities()
+    public function getAllUserActivities($userId)
     {
-        return Activity::where('user_id', Auth::id())->get();
+        return Activity::where('user_id', $userId)->get();
+    }
+
+    public function getUserFolloweeActivities($userId)
+    {
+        return Activity::where('follower_id', $userId)
```
Comment (nguyenthanhtung88) [inline app/Activity.php]: My suggestion for this logic:
```
// Get followee ids by current userId
$followeeIds = Follow::where('follower_id', $userId)->lists('followee_id');

// Get activities of followees
$activities = Activity::with('user')->whereIn('user_id', $followeeIds);
```
=> I think logic is more clearer, you don't have to use `leftJoin` in this case, try to use `Eloquent`, it's much powerful

**Comment by nguyenthanhtung88** [comment]
OK

**Pull request merged**